### PR TITLE
fix: pre-resolve Wikipedia titles at build time to fix iOS image loading

### LINKED ISSUE #97: Image fetch broke for some species again
image

image

Seems like hyphenated names etc are affected. Probably regressed due to recent 404 suppression fixes

### PR DESCRIPTION
**Pull request by jlian**
## Summary

Fixes #97 — image fetch broke for some species on iOS.

Replaces the runtime multi-strategy Wikipedia lookup (up to 6 fetches per species) with a build-time hydration approach that bakes pre-resolved Wikipedia article titles into `taxonomy.json`.

### Changes

- **`scripts/hydrate-wiki-titles.mjs`** — New script that bulk-resolves Wikipedia titles via Wikidata SPARQL, with REST API fallback and retry logic. Achieves 100% coverage (11,167/11,167 species).
- **`src/lib/taxonomy.ts`** — Parses optional 4th element (`wikiTitle`) and exports `getWikiTitle()`.
- **`src/lib/wikimedia.ts`** — Simplified: uses `getWikiTitle()` for single-fetch lookup via REST API. Species not in taxonomy return `undefined` immediately (no runtime fallback needed since hydration achieves 100% coverage). Distinguishes 404 (permanent miss) from transient errors (500/502/503) to avoid caching temporary failures.
- **`src/hooks/use-bird-image.ts`** — Removed iOS blob fetch workaround; sets URL directly.
- **`src/lib/taxonomy.json`** — Hydrated with pre-resolved Wikipedia titles as 4th element.
- **`src/__tests__/wikimedia.test.ts`** — Rewritten to mock `getWikiTitle`; tests taxonomy-hit (single fetch), transient error retry, and edge cases.
- **`src/components/pages/BirdDexPage.tsx`** — Detail view uses full-resolution `originalimage` instead of thumbnail.
- **`package.json`** — Added `taxonomy:hydrate-wiki` npm script.
- **`scripts/validate-wikipedia.mjs`** & **`src/__tests__/ebird-csv.test.ts`** — Updated to remove stale `WIKI_OVERRIDES` references.
- **`e2e/seeded-app.spec.ts`** — Added e2e test for Wikipedia image loading on detail view.

### Verification

- `npm run taxonomy:hydrate-wiki` → 11,167/11,167 species hydrated (100%)
- `npm run taxonomy:validate-wiki` → 11,167/11,167 species validated (100%)
- `npx vitest run` → 428/428 tests passing
- `npx playwright test` → all e2e tests passing

### CODE REVIEW

File: src/lib/wikimedia.ts [hunk 1]
```diff
@@ -52,8 +38,7 @@ async function fetchSummary(title: string): Promise<FetchSummaryResult> {
     const normalizedTitle = title.replace(/ /g, '_')
     const encodedTitle = encodeURIComponent(normalizedTitle)
     const res = await fetch(
-      `https://en.wikipedia.org/w/api.php?action=query&format=json&formatversion=2&origin=*&redirects=1&prop=extracts|pageimages|info&inprop=url&exintro=1&explaintext=1&piprop=thumbnail|original&pithumbsize=600&titles=${encodedTitle}`,
-      { headers: { 'Api-User-Agent': 'BirdDex/1.0 (bird identification app)' } }
+      `https://en.wikipedia.org/w/api.php?action=query&format=json&formatversion=2&origin=*&redirects=1&prop=extracts%7Cpageimages%7Cinfo&inprop=url&exintro=1&explaintext=1&piprop=thumbnail%7Coriginal&pithumbsize=600&titles=${encodedTitle}`
```
[the remark this answers is not in the document]
Comment (jlian) [reply src/lib/wikimedia.ts]: Added Api-User-Agent header in 11e2f41.

File: src/lib/wikimedia.ts [hunk 2]
```diff
@@ -1,125 +1,80 @@
...
     const res = await fetch(
-      `https://en.wikipedia.org/w/api.php?action=query&format=json&formatversion=2&origin=*&redirects=1&prop=extracts|pageimages|info&inprop=url&exintro=1&explaintext=1&piprop=thumbnail|original&pithumbsize=600&titles=${encodedTitle}`,
+      `https://en.wikipedia.org/api/rest_v1/page/summary/${encoded}`,
       { headers: { 'Api-User-Agent': 'BirdDex/1.0 (bird identification app)' } }
     )
-    if (!res.ok && res.status !== 404) {
-      console.warn(`[wikimedia] Unexpected summary response ${res.status} for "${title}"`)
-      return { kind: 'error' }
-    }
     if (!res.ok) return { kind: 'miss' }
```
[the remark this answers is not in the document]
Comment (jlian) [reply src/lib/wikimedia.ts]: Fixed in c681f6b — `fetchSummary` now returns `{ kind: "miss" }` only for 404 (permanent miss). Other non-OK statuses (500, 502, 503) return `{ kind: "error" }` which skips the cache, allowing retry on subsequent calls. Added a test verifying transient errors are not cached.
